optuna/_gp/optim_mixed.py

```diff
@@ -0,0 +1,309 @@
+from __future__ import annotations
+
+import math
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from optuna._gp.acqf import AcquisitionFunctionParams
+from optuna._gp.acqf import eval_acqf_no_grad
+from optuna._gp.acqf import eval_acqf_with_grad
+from optuna._gp.search_space import normalize_one_param
+from optuna._gp.search_space import sample_normalized_params
+from optuna._gp.search_space import ScaleType
+from optuna.logging import get_logger
+
+
+if TYPE_CHECKING:
+    import scipy.optimize as so
+else:
+    from optuna import _LazyImport
+
+    so = _LazyImport("scipy.optimize")
+
+_logger = get_logger(__name__)
+
+
+def _local_search_continuous(
```

Comment:
I think `_gradient_ascent` is simpler.

```suggestion
def _gradient_ascent(
```